### PR TITLE
Build: abacus can compile pw version with cmake

### LINKED ISSUE #1421: abacus_pw.exe link error without -o2
### Expected behavior

_No response_

### To Reproduce

cmake -DBUILD_TESTING=1 -DCMAKE_BUILD_TYPE=Debug

### Environment

OS: Ubuntu 22.04
Compiler: gcc 11.2
Dependencies: OpenBLAS, ELPA

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Qianruipku**
1. It is added to the workflows of github
2. delete temporary tests in src_pw
3. fix issue #1393

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +22,11 @@ option(BUILD_TESTING "Build ABACUS unit tests" OFF)
 option(GENERATE_TEST_REPORTS "Enable test report generation" OFF)
 option(INFO "Enable gathering of math library information" OFF)
 
-set(ABACUS_BIN_NAME abacus)
+if(ENABLE_LCAO)
+  set(ABACUS_BIN_NAME abacus)
+else()
+  set(ABACUS_BIN_NAME abacus_pw)
+endif()
```
Comment (caic99) [inline CMakeLists.txt:29]: What's the point assigning different binary name?

File: source/module_gint/CMakeLists.txt [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-    grid_meshk.cpp
-    grid_technique.cpp
-)
-
-add_library(
-    gint
-    OBJECT
-    ${objects}
-)
+if(ENABLE_LCAO)
```
Comment (caic99) [inline source/module_gint/CMakeLists.txt]: For these libraries only in LCAO basis, you should keep it out in the root CMakeLists.txt .

File: CMakeLists.txt [hunk 3]
```diff
@@ -314,27 +323,32 @@ target_link_libraries(${ABACUS_BIN_NAME}
...
+  lcao
+  orb
+  gint
+  mrrr
+  pdiag
+  genelpa
+  rpa
+  )
+endif()
+
```
Comment (caic99) [inline CMakeLists.txt]: Please note the indentations.

**Review by caic99 — CHANGES_REQUESTED** [review]
@baixiaokuang @mohanchen As we mentioned in the last meeting, further discussion on whether we need a pw-basis-only abacus executable and its necessity should be carried out.
@Qianruipku If there are any updates on this topic, I'll let you know.

File: CMakeLists.txt (hunk 3 above)
Comment (Qianruipku) [reply CMakeLists.txt]: Ok

**Review by caic99 — APPROVED** [review]
After conversation with @baixiaokuang , we think this solution is able to keep the development of planewave module independent with LCAO.

**Pull request merged**